src/FileUploadWidget.js

```diff
@@ -1,47 +1,264 @@
+
+      if(fileSignature == null) {
+        console.log(file);
+        console.log("The file signature for file could not be located.");
+      }
+
+      console.log(fileSignature);
+
+      try {
+        fields = Object.keys(fileSignature["fields"]);
+        console.log("handleUpload() Object keys loaded");
+      } catch (error) {
+        fields = [];
+        console.log("Yikes, no good");
+      }
+
+      for (const key of fields) {
+        console.log(key + " = " + fileSignature["fields"][key]);
+        formData.append(key, fileSignature["fields"][key]);
+      }
+      console.log("fieldName:" + fieldName);
+      console.log("file:" + file);
+      console.log("key:" + fileSignature["fields"]["key"]);
+      formData.append('file', file, fileSignature["fields"]["key"]);
+
+      const request = new XMLHttpRequest();
+
+      request.open('POST', fileSignature["url"]);
+
+      // Should call the progress method to update the progress to 100% before calling load
+      // Setting computable to false switches the loading indicator to infinite mode
+      request.upload.onprogress = (e) => {
+          progress(e.lengthComputable, e.loaded, e.total);
+      };
+
+      // Should call the load method when done and pass the returned server file id
+      // this server file id is then used later on when reverting or restoring a file
+      // so your server knows which file to return without exposing that info to the client
+      request.onload = function() {
+          if (request.status >= 200 && request.status < 300) {
+              // the load method accepts either a string (id) or an object
+              load(request.responseText);
+          }
+          else {
+              // Can call the error method if something is wrong, should exit after
+              error('oh no');
+          }
+      };
+
+      request.send(formData);
+
+      // Should expose an abort method so the request can be cancelled
+      return {
+          abort: () => {
+              // This function is entered if the user has tapped the cancel button
+              request.abort();
+
+              // Let FilePond know the request has been cancelled
+              abort();
+          }
+      };
+  }
+
+  withQuery (url, params) {
+    let query = Object.keys(params)
+      .filter(k => params[k] !== undefined)
+      .map(k => encodeURIComponent(k) + '=' + encodeURIComponent(params[k]))
+      .join('&');
+    url += (url.indexOf('?') === -1 ? '?' : '&') + query;
+    return url;
+  }
+
+  parseSignatureResponse (res) {
+    console.log("parseSignatureResponse: ");
+    console.log(res["filename"]);
+    console.log(res["creds"]["fields"] || "Invalid response")
+    this.fileList.push(res);
+
+    const value = this.fileList.length
       ? JSON.stringify(
-          fileItems.map(f => `${f.filename} - ${f.fileSize} bytes`)
+          this.fileList.map(f => `${f.creds.fields.key}`)
         )
       : false;
+
     this.props.onChange(value);
```

Comment:
Are we certain this is the only place we are updating this value? Just want to make sure it works with multiple file uploads/deletes/etc.